src/words.ts

```diff
@@ -1,10 +1,43 @@
 import {mmod} from 'xen-dev-utils';
+import {TimeMonzo} from './monzo';
+import {ZERO} from './utils';
 
 /**
  * Step count vector indexed by strings representing distinct step sizes.
  */
 export type StepVector = Record<string, number>;
 
+/**
+ * Standard letters representing steps of positive size, arranged from smallest to largest indexed by variety.
+ */
+const POSITIVE_STEP_LETTERS_BY_VARIETY = [
+  [],
+  ['x'],
+  ['s', 'L'],
+  ['s', 'n', 'L'],
```

Comment:
Using either **M** or **m** for ternary scales seems too entrenched at this point, so I'm in favor of `M` in place of `n` for ternary scales.

I myself have used **x**, **y**, **z**, **X**, **Y**, **Z**, or subscripts on these letters for unordered step sizes, and  **0** for zero steps. In some contexts like [MOS substitution](https://en.wikipedia.org/wiki/MOS_substitution), I've used **X** in an otherwise ordered-step-size context for step sizes that can become anything (not necessarily substituting **X** for a fixed letter, but rather substituting individual letters of patterns for the slots represented by **X**; for an example, substituting the **X**s of the 5**L**4**X** MOS string **LXLXLXLXL** with repetitions of **Ms** results in right-hand diasem, **LMLsLMLsL**). For your system, reserving `U`, `V`, `W `(standing for "what"), `X`, `Y`, and `Z` might work for unknown step sizes mixed in with otherwise ordered steps, although *u*, *v*, and *w* seems to often stand for words in academic literature. Maybe we could eventually enable declaring multi-character letters, or at least just subscripts (e.g. `x1`).